### DISCUSSION TITLE
How can I disable Hyprland from using discrete gpu?

### DISCUSSION THREAD

**Discussion by So1aric** [hyprwm/Hyprland#7035] [Q&A]
I want to disable the Nvidia gpu for lower power comsumption. How can I tell Hyprland to only use integerated gpu?
I enabled rtd3 like what the arch wiki said, but my card isn't turned off. I guess maybe it's hyprland that blocks it from turnning off?

**Comment by izmyname** [comment]


**Reply by So1aric** [reply]
THanks for the instruction but well, I set that in .config/hypr/hyprland.conf, but it doesn't make any changes (after reboot, in nvidia-smi it shows that hyprland still uses the dGPU). Does the shell I'm using take count? I set fish as the user's shell.

**Reply by So1aric** [reply]
Actually I've already add these. And they seem to work, that if I exit Hyprland, after ~5seconds the gpu is turned off. So I guess that it's hyprland that block the gpu from turning off.

**Reply by So1aric** [reply]
By the way, is it normal that my cards' index (under /dev/dri) are card1 and card2 instead of 'card0 and card1' in the wiki's example?

**Reply by So1aric** [reply]
```env = AQ_DRM_DEVICES,/dev/dri/card2:/dev/dri/card1```
I add  this line to hyprland.conf. By the way, Hyprland still uses dGPU even if I set the AQ_DRM_DEVICES to a path that doesn't exist. I expect it to crash or something (don't know is it correct or not)

**Reply by So1aric** [reply]
Doesn't work :(

**Reply by So1aric** [reply]
4060 mobile. the output is: 
```
Runtime D3 status:          Enabled (fine-grained)
Video Memory:               Active

GPU Hardware Support:
 Video Memory Self Refresh: Supported
 Video Memory Off:          Supported

S0ix Power Management:
 Platform Support:          Supported
 Status:                    Disabled
```

**Reply by So1aric** [reply]
That's alright, thanks for you help!

**Comment by gulafaran** [comment]
for what its worth i had same issue when it was still wlroots based on this 4080 dgpu

**Reply by So1aric** [reply]
What about now? I heard that hyprland already dropped wlroots dependency, right?

**Reply by gulafaran** [reply]
yeah what i meant is my nvidia dgpu has been awoken by both wlroots and now aquamarine even with forcing various env vars. kde/kwin does something different as it can keep it off

**Comment by So1aric** [comment]
I switch to hyprland-git, and it successfully uses iGPU!
But still, something blocks nvidia from turning off. When I'm in tty, the dGPU automatically turned off, but in Hyprland, although there's no process using the gpu, the card wasn't turned off.

**Reply by So1aric** [reply]
Oops. It's mpvpaper running that cause the block. My fault.
But nvidia-smi doesn't list it under the 'Processes' section....Is it because nvidia-smi only list graphic applications by default?

**Reply by gulafaran** [reply]
thats interesting, so i might be having some rogue thing keeping my gpu awake aswell that isnt listed or hyprland? :o how did you figure that out?

**Reply by So1aric** [reply]
I was thinking that will there be anything using the gpu, and I recalled that I saw mpvpaper depend on cuda in some versions, so I just kill it and, as luck would have it, the card got turned off.

**Reply by So1aric** [reply]
I'm not totally sure about that. I'm trying more tests.

**Reply by izmyname** [reply]
Interesting enough, my novideo gpu can suspend only, if the laptop is on the battery mode - but it wasn't the case, until I reinstalled my arch recently. Before that, my dgpu could suspend even when my laptop was plugged in.

**Comment by So1aric** [comment]
Oops I see you mentioned about this in the aquamarine repo
https://github.com/hyprwm/Hyprland/pull/6608#issuecomment-2241713208
